quix-frontend/client/src/lib/sql-autocomplete/adapter/sql-autocpmplete-adapter-utills.ts

```diff
@@ -0,0 +1,304 @@
+  name: string;
+  dataType: any;
+}
+
+export function getObjectChildren(obj: Record<string, any>, parentName = ''): ObjectChild[] {
+  const children: ObjectChild[] = [];
+
+  for (const key in obj) {
+    if (typeof obj[key] === 'object' && obj[key] !== null) {
+      const childName = parentName ? `${parentName}.${key}` : key;
+      children.push({ name: childName, dataType: obj[key] });
+      children.push(...getObjectChildren(obj[key], childName));
+    } else {
+      const childName = parentName ? `${parentName}.${key}` : key;
+      children.push({ name: childName, dataType: obj[key] });
+    }
+  }
+
+  return children.map(child => {
+    child.name = child.name.replace(/\.dataType$/, '');
+    return child;
+  });
+}
+
+const enum StartingOptions {
+  row = "row(",
+  map = "map(",
+  array = "array(",
+  timestamp = "timestamp(",
+}
+
+const enum SpecialCharacters {
+  openParenthesis = "(",
+  closedParenthesis = ")",
+  comma = ',',
+  space = ' ',
+}
+
+export function trinoToJs(trinoObjectAsString: string, index: number):  object|string {
+  if (startsWithRow(trinoObjectAsString, index)) {
+    index += 4; // jump after "row("
+    return processRow(trinoObjectAsString, index);
+  }
+  if (trinoObjectAsString.substring(index, index + 4) === StartingOptions.map) {
+    index = findClosingParentheses(trinoObjectAsString, index);
+    return "map";
+  }
+
+  if (trinoObjectAsString.substring(index, index + 6) === StartingOptions.array) {
+    index = findClosingParentheses(trinoObjectAsString, index);
+    return "array";
+  }
+    return trinoObjectAsString;
+}
+
+export function processRow(trinoObjectAsString: string, index: number): object {
+    const finalObject: object = {};
+    let key = "";
+    let value = "";
+    let objectToInsert;
+    let firstword = true;
+  
+    while (index <= trinoObjectAsString.length) {
+      switch (trinoObjectAsString.charAt(index)) {
+        case SpecialCharacters.openParenthesis: {
+          const start = findBeginningOfWord(trinoObjectAsString, index);
+          const result = handleStartingOptions(trinoObjectAsString, start, index);
+          objectToInsert = result.objectToInsert;
+          finalObject[key] = result.type;
+          index = result.newIndex;
+          break;
+        }
+        case SpecialCharacters.closedParenthesis: {
+          checkKey(key, index);
+          finalObject[key] = objectToInsert || value;
+          return finalObject;
+        }
+        case SpecialCharacters.comma: {
+          if (value === "") {
+            throw new Error(`Error at index: ${trinoObjectAsString.substring(index-5,index)}, type expected before comma`);
+          }
+          finalObject[key] = objectToInsert || value;
+          key = "";
+          value = "";
+          objectToInsert = undefined;
+          firstword = true;
+          break;
+        }
+        case SpecialCharacters.space: {
+          firstword = isComma(trinoObjectAsString, index);
```

Comment:
consider again if isComma function is needed.
`isFirstWord = trinoObjectAsString[index-1] === SpecialCharacters.comma;`

if you think it needed - simplify the isComma function. it should be very simple function that gets only a character and return true/false